traitlets/traitlets.py

```diff
@@ -667,9 +681,27 @@ def __get__(self, inst, cls):
             return self
         return types.MethodType(self.func, inst)
 
+class ObserveHandler(EventHandler):
+
+    def __init__(self, names=None):
+        if names is None:
+            self.names=[None]
+        else:
+            self.names = names
+
     def instance_init(self, inst):
+        meth = types.MethodType(self.func, inst)
         for name in self.names:
-            inst.observe(types.MethodType(self.func, inst), name=name)
+            inst.observe(meth, name)
+
+class ValidateHandler(EventHandler):
+
+    def __init__(self, name):
+        self._name = name
+    
+    def instance_init(self, inst):
+        meth = types.MethodType(self.func, inst)
+        inst.register_validator(meth, self._name)
```

Comment:
Sure, that would work.  In addition to checking to see if there is a registered validator already, also check for a magic method name.